### PR TITLE
dashboard-e2e-job: get mgr logs as an artifact

### PR DESCRIPTION
**Pull request by nizamial09**
archive the logs as an artifact in both pr and nightly jobs of
dashboard frontend e2es.

Screenshot 2022-03-02 143840

Signed-off-by: Nizamudeen A <nia@redhat.com>

### CODE REVIEW

**Comment by nizamial09** [comment]
sample run: https://jenkins.ceph.com/job/ceph-dashboard-pull-requests/10139/

File: ceph-api-nightly/config/definitions/ceph-api-nightly.yml [hunk 1]
```diff
@@ -89,3 +89,8 @@
     publishers:
       - email:
           recipients: ceph-qa@ceph.io
+
+      - archive:
+          artifacts: 'build/out/mgr.x.log'
```
Comment (epuertat) [inline ceph-api-nightly/config/definitions/ceph-api-nightly.yml:94]: What about this?
```suggestion
          artifacts: 'build/out/mgr.*.log'
```
And perhaps in the future we'd need to add every log... But let's wait till then.

**Review by epuertat — APPROVED** [review]
LGTM!

Shouldn't we add this to ceph-api ones too (the cephadm would be harder since the logs are inside VMs, so we would need to make those volumes accessible from the outside)?

BTW @djgalloway along with this PR it makes sense to install/enable the [artifact compression](https://plugins.jenkins.io/compress-artifacts/) (text logs are easily compressible).

**Comment by nizamial09** [comment]
> we add this to ceph-api ones

will add it too. 

> make those volumes accessible from the outside

or at the end of the test we can right the mgr container log on a file and point that file on the artifact?

**Comment by djgalloway** [comment]
> BTW @djgalloway along with this PR it makes sense to install/enable the [artifact compression](https://plugins.jenkins.io/compress-artifacts/) (text logs are easily compressible).

That should get added here: https://github.com/ceph/ceph-build/blob/master/ansible/examples/master.yml#L12.

Let's go ahead and add that then configure that plugin in the JJB config in this PR.

**Comment by nizamial09** [comment]
> Let's go ahead and add that then configure that plugin in the JJB config in this PR.

@djgalloway I added it to the master.yml file but I am not sure how to configure the plugin in JJB. Could you help me with that? I couldn't find a documentation on how to configure the compress-artifacts.

**Comment by epuertat** [comment]
> So I installed it but the plugin compresses ALL artifacts for ALL jobs. It can't be enabled per-job. Is it possible to compress the logs in the job before archiving them?

@djgalloway What's the downside of it? When you click on those links, does the browser uncompress those or you have to download+open+etc.? 😞

What file system are we using for these nodes? Can't we have transparent compression at file system level? Call me mad, but can't we use [Ceph/S3](https://plugins.jenkins.io/artifact-manager-s3/) for that (I've heard it supports compression :P)?

**Comment by djgalloway** [comment]
> @djgalloway What's the downside of it? When you click on those links, does the browser uncompress those or you have to download+open+etc.? disappointed

I guess not.  It's been enabled for most of today and I can click filenames in the Web UI and download them no problem.

**Comment by epuertat** [comment]
@djgalloway if we all are ok with compressing artifacts, could we get this merged plz? Thank you!

**Pull request merged**